### PR TITLE
Version info should be passed  to the boot up log message

### PR DESCRIPTION
**Pull request by rpothier**
### Desired Outcome

conjur-authn-k8s-client release artifact reports wrong version

### Implemented Changes

The Version from the automated release is used in the startup logs.

Also updated the changelog as the next version should be version 0.24.1.

### Connected Issue/Story

CyberArk internal issue ID: CNJR-770

### Definition of Done
*At least 1 todo must be completed in the sections below for the PR to be
merged.*

#### Changelog

- [x] The CHANGELOG has been updated, or
- [ ] This PR does not include user-facing changes and doesn't require a
  CHANGELOG update

#### Test coverage

- [ ] This PR includes new unit and integration tests to go with the code
  changes, or
- [x] The changes in this PR do not require tests

#### Documentation

- [ ] Docs (e.g. `README`s) were updated in this PR
- [ ] A follow-up issue to update official docs has been filed here: [insert issue ID]
- [x] This PR does not require updating any documentation

#### Behavior

- [ ] This PR changes product behavior and has been reviewed by a PO, or
- [ ] These changes are part of a larger initiative that will be reviewed later, or
- [x] No behavior was changed with this PR

#### Security

- [ ] Security architect has reviewed the changes in this PR,
- [ ] These changes are part of a larger initiative with a separate security review, or
- [x] There are no security aspects to these changes

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -148,6 +150,9 @@ RUN git clone https://github.com/ztombol/bats-support /bats/bats-support && \
 RUN wget https://github.com/mikefarah/yq/releases/latest/download/yq_linux_amd64 -O /usr/bin/yq && \
     chmod +x /usr/bin/yq
 
+# Update go.mod due to CVE-2022-41723
+RUN sed -i "s|v0.4.1-0.20230214201333-88ed8ca3307d|v0.7.0|g" /usr/local/go/src/go.mod
```
[the remark this answers is not in the document]
Comment (rpothier) [reply Dockerfile:154]: Yes, this is a short term solution until the base image is updated.

File: Dockerfile [hunk 2]
```diff
@@ -125,7 +127,7 @@ LABEL description="The authentication client required to expose secrets from a C
 
 # =================== CONTAINER FOR HELM TEST ===================
 
-FROM golang:alpine as k8s-cluster-test
+FROM golang:1.20-alpine as k8s-cluster-test
```
Comment (szh) [inline Dockerfile:130]: Why is this necessary? I don't think we're targeting Go 1.20 in any of the other images

Comment (rpothier) [reply Dockerfile:130]: Currently `golang:alpine` and `golang:1.20-alpine` are the same image, so there is currently no change to the image.
#502 updated golang in the Dockerfiles to 1.19, a fixed version, and golang:alpine will change over time.
We could use 1.19 however that version is older and that won't help the security issues.
This change is not needed for the version info.

**Review by szh — APPROVED** [review]
LGTM

**Pull request merged**